gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobScheduler.java

```diff
@@ -82,6 +106,146 @@ public GobblinHelixJobScheduler(Properties properties, HelixManager helixManager
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void switchMetricContext(MetricContext context) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public StandardMetricsBridge.StandardMetrics getStandardMetrics() {
+    return metrics;
+  }
+
+  private class InnerStandardMetrics implements StandardMetrics {
+
+    private final ContextAwareCounter numJobsLaunched;
+    private final ContextAwareCounter numJobsCompleted;
+    private final ContextAwareCounter numJobsCommitted;
+    private final ContextAwareCounter numJobsFailed;
+    private final ContextAwareCounter numJobsCancelled;
+    private final ContextAwareGauge<Integer> numJobsRunning;
+
+    public InnerStandardMetrics(final GobblinHelixJobScheduler jobScheduler) {
+      this.numJobsLaunched = jobScheduler.getMetricContext().contextAwareCounter(JobExecutionLauncher.StandardMetrics.NUM_JOBS_LAUNCHED_COUNTER);
+      this.numJobsCompleted = jobScheduler.getMetricContext().contextAwareCounter(JobExecutionLauncher.StandardMetrics.NUM_JOBS_COMPLETED_COUNTER);
+      this.numJobsCommitted = jobScheduler.getMetricContext().contextAwareCounter(JobExecutionLauncher.StandardMetrics.NUM_JOBS_COMMITTED_COUNTER);
+      this.numJobsFailed = jobScheduler.getMetricContext().contextAwareCounter(JobExecutionLauncher.StandardMetrics.NUM_JOBS_FAILED_COUNTER);
+      this.numJobsCancelled = jobScheduler.getMetricContext().contextAwareCounter(JobExecutionLauncher.StandardMetrics.NUM_JOBS_CANCELLED_COUNTER);
+      this.numJobsRunning = jobScheduler.getMetricContext().newContextAwareGauge(JobExecutionLauncher.StandardMetrics.NUM_JOBS_RUNNING_GAUGE,
+          new Gauge<Integer>() {
+            @Override public Integer getValue() {
+              return (int)(InnerStandardMetrics.this.numJobsLaunched.getCount() -
+                  InnerStandardMetrics.this.numJobsCompleted.getCount());
+            }
+          });
+    }
+
+    @Override
+    public String getName() {
+      return GobblinHelixJobScheduler.class.getName();
+    }
+
+    @Override
+    public Collection<ContextAwareGauge<?>> getGauges() {
+      return Collections.singleton(numJobsRunning);
+    }
+
+    @Override
+    public Collection<ContextAwareCounter> getConters() {
+      List<ContextAwareCounter> counters = Lists.newArrayList();
+      counters.add(numJobsLaunched);
+      counters.add(numJobsCompleted);
+      counters.add(numJobsCommitted);
+      counters.add(numJobsFailed);
+      counters.add(numJobsCancelled);
+      return counters;
+    }
+
+    @Override
+    public Collection<ContextAwareMeter> getMeters() {
+      return null;
+    }
+
+    @Override
+    public Collection<ContextAwareTimer> getTimers() {
+      return null;
+    }
+
+    @Override
+    public Collection<ContextAwareHistogram> getHistograms() {
+      return null;
+    }
+  }
+
+  @AllArgsConstructor
+  private class InnerJobListener extends AbstractJobListener {
```

Comment:
Instead of `Inner` maybe use a name like `Metrics` or `MetricsTracking` to indicate what this listener does.